packages/algob/src/lib/algo-operator.ts

```diff
@@ -60,4 +70,56 @@ export class AlgoOperatorImpl implements AlgoOperator {
       confirmedRound: txConfirmation[confirmedRound]
     };
   }
+
+  async deployASC (programb64: string, scParams: object, flags: ASCDeploymentFlags
+  ): Promise<ASCInfo> {
+    const program = new Uint8Array(Buffer.from(programb64, "base64"));
+    const encoder = new TextEncoder();
+    const lsig = algosdk.makeLogicSig(program, scParams);
+
+    const params = await getSuggestedParamsWithUserDefaults(this.algodClient, flags);
+
+    // ASC1 signed by funder
+    lsig.sign(flags.creator.sk);
+    const funder = flags.creator.addr;
+    const contractAddress = lsig.address();
+    // const receiver = funder;
+
+    // Fund smart contract
+    console.log("Funding Contract:", contractAddress);
+
+    const tran = algosdk.makePaymentTxnWithSuggestedParams(funder, contractAddress,
+      flags.microAlgo, flags.closeToRemainder,
+      flags.note ? encoder.encode(flags.note) : undefined,
+      params);
+
+    const signedTxn = tran.signTxn(flags.creator.sk);
+
+    const tranInfo = await this.algodClient.sendRawTransaction(signedTxn).do();
+
+    const confirmedTxn = await this.algodClient.pendingTransactionInformation(tranInfo.txId).do();
```

Comment:
I think this doesn't wait for tx confirmation.
Does it call `waitForConfirmation`?

Reply:
No, this is what i found in algorand docs for making a payment transaction

Reply:
But the pending transaction will still have to be waited to be confirmed. It should be the same for all transactions.

Reply:
Yes , added